lib/modules/platform/github/massage-markdown-links.ts

```diff
@@ -10,13 +10,25 @@ interface UrlMatch {
   replaceTo: string;
 }
 
+/*
+explaining the long regex
+ http..github.com/ = (?:https?:\/\/)?(?:www\.)?(?:to)?github\.com\/
+ org/repo/ = (?<org>[-_.\w\d]+)\/(?<repo>[-_.\w\d]+)\/
+ pull/14476 = (?:discussions|issues|pull)\/(?<number>\d+)
+ #comment-123 = (?:#[-_.\w\d]+)?
+ according to: https://github.com/dead-claudia/github-limits
+*/
 const urlRegex =
-  /(?:https?:)?(?:\/\/)?(?:www\.)?(?<!api\.)(?:to)?github\.com\/[-_a-z0-9]+\/[-_a-z0-9]+\/(?:discussions|issues|pull)\/[0-9]+(?:#[-_a-z0-9]+)?/i; // TODO #12872 (?<!re) after text not matching
+  /(?:https?:\/\/)?(?:www\.)?(?:to)?github\.com\/(?<org>[-\w\d]+)\/(?<repo>[-_.\w\d]+)\/(?:discussions|issues|pull)\/(?<number>\d+(?:#[-_.\w\d]+)?)/i; // TODO #12872 (?<!re) after text not matching
```

Comment:
i think we can now use RE2, as you removed the look-ahead regex.

Reply:
I am not familiar with that, can you explain what is needed here?

Reply:
```suggestion
  regex(/(?:https?:\/\/)?(?:www\.)?(?:to)?github\.com\/(?<org>[-\w\d]+)\/(?<repo>[-_.\w\d]+)\/(?:discussions|issues|pull)\/(?<number>\d+(?:#[-_.\w\d]+)?)/i);
```

this should work now

Reply:
do you want me to change the syntax to the new regex syntax RE2?

Reply:
this is a problem for global regex, i tried it fails after that

Reply:
the usage here has `gi`
```
      const globalUrlReg = new RegExp(urlRegex, 'gi');
      const urlMatches = [...tree.value.matchAll(globalUrlReg)];
```

Reply:
this code was there before i even started changing anything, 
i try changing it to 
```
     const globalUrlReg = regEx(urlRegex, 'gi');
      const urlMatches = [...tree.value.matchAll(globalUrlReg)];
```
but then 16 tests fail